android/core/src/main/java/com/stadiamaps/ferrostar/core/annotation/SpeedSerializationAdapter.kt

```diff
@@ -18,7 +18,7 @@ class SpeedSerializationAdapter : JsonAdapter<Speed>() {
         is Speed.NoLimit -> writer.name("none").value(true)
         is Speed.Unknown -> writer.name("unknown").value(true)
         is Speed.Value ->
-            writer.name("value").value(speed.value).name("unit").value(speed.unit.text)
+            writer.name("speed").value(speed.value).name("unit").value(speed.unit.text)
```

Comment:
@ahmedre I believe the actual key should be `"speed"` when there's a value. E.g.

```json
{
  "speed": 56,
  "unit": "km/h"
},
```

Let me know if there are additional cases. Definitely a bit of an obnoxious json object 😄.

Also, do we want to try catch the annotations parsing? On iOS we just convert a json parsing error to nil/null for annotations and an onError callback so that developers can log the failure. We figured better to let navigation succeed even in the case of a degraded/malformed annotations. See https://github.com/stadiamaps/ferrostar/blob/61e299a66a1afc02c046ff52e1d6ff48a1f3cffa/apple/Sources/FerrostarCore/Annotations/AnnotationPublisher.swift#L70-L75

Reply:
yes, you are right - my mistake! agree re: try/catching the parsing with an error callback 👍